cms/djangoapps/modulestore_migrator/rest_api/v1/views.py

```diff
@@ -571,31 +573,63 @@ def get(self, request: Request):
         """
         Handle the migration info `GET` request
         """
-        source_key = request.query_params.get("source_key")
-        target_key = request.query_params.get("target_key")
-        target_collection_key = request.query_params.get("target_collection_key")
-        task_uuid = request.query_params.get("task_uuid")
-        is_failed: str | bool | None = request.query_params.get("is_failed")
-        if not target_key:
+        target_key: LibraryLocatorV2 | None
+        if target_key_param := request.query_params.get("target_key"):
+            try:
+                target_key = LibraryLocatorV2.from_string(target_key_param)
+            except InvalidKeyError:
+                return Response({"error": f"Bad target_key: {target_key_param}"}, status=400)
+        else:
             return Response({"error": "Target key cannot be blank."}, status=400)
-        try:
-            target_key_parsed = LibraryLocatorV2.from_string(target_key)
-        except InvalidKeyError as e:
-            return Response({"error": str(e)}, status=400)
+        target_collection_key = request.query_params.get("target_collection_key")
+        source_key: SourceContextKey | None
+        if source_key_param := request.query_params.get("source_key"):
+            try:
+                source_key = CourseLocator.from_string(source_key_param)
+                try:
+                    source_key = LibraryLocator.from_string(source_key_param)
+                except InvalidKeyError:
+                    return Response({"error": f"Bad source: {source_key_param}"}, status=400)
+            except InvalidKeyError:
+                return Response({"error": f"Bad source: {source_key_param}"}, status=400)
```

Comment:
~You probably want to move `LibraryLocator.from_string(...)` call inside `except` part of the first try catch.~ I see that @ormsbee has already commented for the same lines. (For some reason the comment is not visible in diff page).